contracts/test/PoolFDTLosses.t.sol

```diff
@@ -0,0 +1,314 @@
+
+            claimInfo1 = pat.claim(address(pool), address(custom_loan), address(dlFactory));
+            claimInfo2 = pam.claim(address(pool2), address(custom_loan), address(dlFactory));
+
+            assertTrue(claimInfo1[6] > 0,        "Loan doesn't has default suffered");
+            assertTrue(pool.poolLosses() > 0, "Pool 1 losses should be greater than 0");
+            withinPrecision(pool.recognizableLossesOf(address(leo)) + pool.recognizableLossesOf(address(liz)),  pool.poolLosses(), 5);
+
+            assertTrue(claimInfo2[6] > 0,         "Loan doesn't has default suffered");
+            assertTrue(pool2.poolLosses() > 0, "Pool 2 losses should be greater than 0");
+            withinPrecision(pool2.recognizableLossesOf(address(lex)) + pool2.recognizableLossesOf(address(lee)), pool2.poolLosses(), 5);
+        }
+
+        // Time warp to withdraw
+        hevm.warp(block.timestamp + pool.lockupPeriod() + 1);
+
+        // Fails to transfer if the losses are > 0.
+        assertTrue(
+            !leo.try_transfer(address(pool), address(liz), 1),
+            "Should not allow Lex to transfer because losses are > 0"
+        );
+
+        assertTrue(
+            !liz.try_transfer(address(pool), address(leo), 1),
+            "Should not allow Lee to transfer because losses are > 0"
+        );
+
+        assertTrue(
+            !lex.try_transfer(address(pool2), address(lee), 1),
+            "Should not allow Leo to transfer because losses are > 0"
+        );
+
+        assertTrue(
+            !lee.try_transfer(address(pool2), address(lex), 1),
+            "Should not allow Liz to transfer because losses are > 0"
+        );
+
+        // LPs is withdrawing funds from the Pool with losses.
+        assertTrue(leo.try_intendToWithdraw(address(pool)),  "Leo Failed to intend to withdraw");
+        assertTrue(liz.try_intendToWithdraw(address(pool)),  "Liz Failed to intend to withdraw");
+        assertTrue(lex.try_intendToWithdraw(address(pool2)), "Lex Failed to intend to withdraw");
+        assertTrue(lee.try_intendToWithdraw(address(pool2)), "Lee Failed to intend to withdraw");
+
+        // Time warp to withdraw window
+        hevm.warp(block.timestamp + globals.lpCooldownPeriod() + 1);
+
+        // Before withdrawing funds.
+        uint256 old_leo_bal = usdc.balanceOf(address(leo));
+        uint256 old_liz_bal = usdc.balanceOf(address(liz));
+        uint256 old_lex_bal = usdc.balanceOf(address(lex));
+        uint256 old_lee_bal = usdc.balanceOf(address(lee));
+
+        assertTrue(leo.try_withdraw(address(pool),  leoDepositAmt), "Leo Failed to withdraw");
+        assertTrue(liz.try_withdraw(address(pool),  lizDepositAmt), "Liz Failed to withdraw");
+        assertTrue(lex.try_withdraw(address(pool2), lexDepositAmt), "Lex Failed to withdraw");
+        assertTrue(lee.try_withdraw(address(pool2), leeDepositAmt), "Lee Failed to withdraw");
```

Comment:
These should all be `user.withdraw()`

Reply:
The only time we should be using `assertTrue(try_)` to assert successful function calls is if we had a corresponding `assertTrue(!try_)` call to assert a failure mode for the same function above

Reply:
Since we have asserted the failure mode for transfers above, we should do this for `try_transfer`

Reply:
We should try withdrawing `userDepositAmt - 1` for leo and lex, and then transferring tokens from
```
leo => liz
lex => lee

then

liz => leo
lee => lex
```

Reply:
This will allow us to assert successful transfers after recognizing losses, and will keep our assertions below the same

Reply:
> Since we have asserted the failure mode for transfers above, we should do this for `try_transfer`

I don't agree on an assertion for a successful `transfer`, because the completion of the test (no revert) is the assertion. The only time positive assertions should be made, IMO, is if the called function cannot revert. Since `transfer` reverts, there is no need to check that it hasn't reverted. `try_transfer` should be exclusively used to test that a call to `transfer` would have reverted, without ending/killing the test when it does.